File: valid_parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if s == "": return True
        stack = []
        for i in s:
            if stack and self.isCompliment(stack[-1],i):
                stack.pop()
            else:
                stack.append(i)
        
        return len(stack) == 0
    
    def isCompliment(self, j, k):
        if j == '[' and k == ']':
            return True
        if j == '{' and k == '}':
            return True
        if j == '(' and k == ')':
            return True
        return False
        

File: valid_parentheses.js
// Check if string of parentheses is valid, using stacks
var isValid = function(s) {
    const isPair = function(left, right){
    if(left === '(' && right === ')'){
        return true
    }else if(left === '[' && right === ']'){
        return true
    }else if(left==='{' && right==='}'){
        return true
    }else{
        return false
    }
    }
    let stack = []
    for(i=0; i<s.length; i++){
        if(stack.length && isPair(stack[stack.length-1], s[i])){
            stack.pop()
        }else{
            stack.push(s[i])
        }
    }
    return stack.length === 0;
}